File: fixtures/scale/java/large/src/cryptobench/ciphers/scytale/ScytaleSuite.java
package cryptobench.ciphers.scytale;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class ScytaleSuite implements CipherSuite {
    @Override
    public String name() {
        return "scytale";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new ScytaleCipher(ScytaleKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : ScytaleVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/dart/large/lib/ciphers/scytale/scytale_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'scytale_cipher.dart';
import 'scytale_key.dart';
import 'scytale_vectors.dart';

class ScytaleSuite implements CipherSuite {
  @override
  String name() => 'scytale';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = ScytaleCipher(ScytaleKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in scytaleSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
